code/game/objects/items/stacks/sheets/glass.dm

```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
 			to_chat(H, span_warning("[src] cuts into your hand!"))
 			H.apply_damage(force*0.5, BRUTE, hit_hand)
 
-/obj/item/shard/attackby(obj/item/I, mob/user, params)
-	if(istype(I, /obj/item/lightreplacer))
-		var/obj/item/lightreplacer/L = I
-		L.attackby(src, user)
-	else if(istype(I, /obj/item/stack/sheet/cloth))
-		var/obj/item/stack/sheet/cloth/C = I
-		to_chat(user, span_notice("You begin to wrap the [C] around the [src]..."))
-		if(do_after(user, 35, target = src))
-			var/obj/item/knife/shiv/S = new /obj/item/knife/shiv
-			C.use(1)
-			to_chat(user, span_notice("You wrap the [C] around the [src] forming a makeshift weapon."))
-			remove_item_from_storage(src)
-			qdel(src)
-			user.put_in_hands(S)
+/obj/item/shard/attackby(obj/item/item, mob/user, params)
+	if(istype(item, /obj/item/lightreplacer))
+		var/obj/item/lightreplacer/lightreplacer = item
+		lightreplacer.attackby(src, user)
+	else if(istype(item, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/cloth = item
+		to_chat(user, span_notice("You begin to wrap the [cloth] around the [src]..."))
+		if(istype(src, /obj/item/shard/plasma))
+			if(do_after(user, 140, target = src))
+				var/obj/item/knife/shiv/plasma/shiv = new /obj/item/knife/shiv/plasma
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/titanium))
+			if(do_after(user, 70, target = src))
+				var/obj/item/knife/shiv/titanium/shiv = new /obj/item/knife/shiv/titanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/plastitanium))
+			if(do_after(user, 70, target = src))
+				var/obj/item/knife/shiv/plastitanium/shiv = new /obj/item/knife/shiv/plastitanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else
+			if(do_after(user, 35, target = src))
+				var/obj/item/knife/shiv/shiv = new /obj/item/knife/shiv
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
```

Comment:
Just have the shard have a var for what it should turn into when you apply cloth, and then we don't need to have all this repeating code. do_after times can also be like that